### ISSUE TITLE
Docker Desktop 4.5.1 missing tray icon

### ISSUE THREAD

**Issue by gpettey** [docker/for-win#12573]
### Actual behavior
Tray icon not present.

### Expected behavior
Tray icon present.

### Information
Reproducible:
* Upgrade from 4.5.0 to 4.5.1 and tray icon is gone.
* Complete uninstall and fresh install of 4.5.1 and tray icon did not appear.
* Complete uninstall and fresh install of 4.4.4, tray icon is present. Upgrade to 4.5.1 and tray icon is gone.

In each case I can access Docker Desktop from the start menu shortcut, but it appears to be spawning a new instance every time I do that (Task Manager eventually shows many "Docker Desktop"s running if I run via the shortcut then close the window then repeat.

Windows 10.0.18363.2094

Diagnostic E47DA4F9-6DED-433F-B321-4B3102BDA456/20220222135227

The "WhaleMenuRedesign" workaround does not solve this.

**Comment by duke1995** [comment]
I have the same problem after upgrading Docker Desktop for Windows to version 4.5.1 (74721), the tray icon is missing.

After closing the main window Docker Desktop keeps running in the background without a way to properly close it.

**Comment by HeCorr** [comment]
I can confirm this. It started happening right after updating from `v4.4.4` to `v4.5.1`.

**Comment by joe0BAB** [comment]
Thanks for reporting! Could you please upload the diagnostics?

**Comment by adpe** [comment]
It's also happening from 4.4.4 to 4.5.0. The workaround is mentioned in the blog post https://www.docker.com/blog/new-docker-menu-improved-release-highlights-with-docker-desktop-4-5/.

Just quit Docker Desktop, then add a `features-overrides.json` file with the following content:
```json
{
  "WhaleMenuRedesign": {
    "enabled": false
  } 
}
```

Depending on your operating system, you will need to place this file in one of the following location:

On Mac: `~/Library/Group Containers/group.com.docker/features-overrides.json`
On Windows: `%APPDATA%\Docker\features-overrides.json`

But for me, the problem is not only the tray icon, also the Electron GUI is not visible, and in the logs it mentioned, that it's running... The tray icon is now visible, after doing the workaround.

Greets
Adrian

**Comment by gpettey** [comment]
As noted in my initial report, the menu was working fine in 4.5.0. It only went away with 4.5.1, which funnily mentions in the release notes "Fixed a crash in the Docker Dashboard which would make the systray menu disappear."

That is a different problem if your GUI is not showing up at all, and your workaround does not work for this problem.

**Comment by trungutt** [comment]
@gpettey To help us debugging the problem, could you please upload the diagnostics? The instruction is here https://docs.docker.com/desktop/windows/troubleshoot/#diagnosing-from-the-terminal

**Comment by duke1995** [comment]
I gathered and uploaded the diagnostics, following the instructions mentioned above.

FYI: The first time I ran the diagnostics Docker Desktop was not running (Diagnostics ID: 94D542C3-4A05-4388-8403-DE4DF728F17F/20220223081905), so I ran the diagnostics a second time with Docker Desktop running (Diagnostics ID: 94D542C3-4A05-4388-8403-DE4DF728F17F/20220223082631).

**Comment by duke1995** [comment]
> It's also happening from 4.4.4 to 4.5.0. The workaround is mentioned in the blog post https://www.docker.com/blog/new-docker-menu-improved-release-highlights-with-docker-desktop-4-5/.

The workaround fixes the problem of the missing tray icon in my case also.

**Comment by HeCorr** [comment]
@adpe's workaround worked for me. Thank you very much!

![image](https://user-images.githubusercontent.com/75134774/155347570-097df945-85a4-46c9-bc95-8afbb9b53bbf.png)

**Comment by lukethegeek** [comment]
Same issue for me when installing / upgrading to 4.5.1.

Edition	Windows 10 Pro
Version	20H2
OS build	19042.1526
Experience	Windows Feature Experience Pack 120.2212.4170.0
WSL2 Backend

Docker GUI / Electron won't open and the system tray icon is gone. When trying to run docker, multiple processes will be spawned and are running in task manager, but there's otherwise no evidence that Docker is running at all.

The "WhaleMenuRedesign" workaround above makes the systray icon appear again. However it just displays as the 'docker isn't running' half whale, and right clicking on it only displays a context menu missing most of the useful options like open dashboard. The GUI / Electron is still completely unable to open.

The only way I could get docker working again was to downgrade to 4.4.4.

**Comment by adpe** [comment]
Hi @lukethegeek 

How do you downgrade without loss of data?

Greets
Adrian

**Comment by lukethegeek** [comment]
Hi Adrian,

> How do you downgrade without loss of data?

I just ended up uninstalling Docker which wiped all of the data. I really wish they gave the option to leave data intact during uninstallation, because uninstalling seems to be the only option until an update with a fix is released.

If you're using WSL2, there is possibly a method whereby you [take a backup of the entire wsl distros / xvhd](https://www.reddit.com/r/bashonubuntuonwindows/comments/hjg4wg/creating_an_image_of_wsl2_to_restore_on_a_new/) then just restore it after installing a working Docker version. But I have no idea how successful this approach is, or whether anything will be lost or broken in the process if downgrading.

**Comment by amari-at4** [comment]
Hi,

I've got the same problem.   After upgrade from 4.4.4 to 4.5.1 the icon in sys tray was missing, then I've applied the workaround and the icon is showed again in the sys tray but the windows for dashboard or setting are unavailable.

**Comment by trungutt** [comment]
@gpettey @duke1995 @amari-at4 @adpe @lukethegeek Could you please try out this build https://desktop-stage.docker.com/win/main/amd64/75084/Docker%20Desktop%20Installer.exe? Hopefully it could bring back the whale icon

**Comment by lukethegeek** [comment]
@trungutt unfortunately it's the same result with build 75084

Docker is running in the task manager, but no GUI or systray icon.

**Comment by trungutt** [comment]
Thanks you @lukethegeek. To help us debugging the problem, could you please upload the diagnostics? The instruction is here https://docs.docker.com/desktop/windows/troubleshoot/#diagnosing-from-the-terminal

**Comment by trungutt** [comment]
I would also encourage others to try out the build above and give back diagnostics. The root cause might not be the same for all

**Comment by adpe** [comment]
Hi @trungutt 

Will try it tomorrow. I've already sent my diagnostics, see issue https://github.com/docker/for-win/issues/12575.

**Comment by lukethegeek** [comment]
@trungutt no problem. Done.

Diagnostics ID:     7A765C8E-24A2-4E4A-8D66-F5C9965122C9/20220224205218 (uploaded)

If you need anything else, please don't hesitate to ask (e.g. installing more test versions / diagnostics / whatever). Happy to do what I can to help.

**Comment by trungutt** [comment]
Thank you @lukethegeek. Could you please confirm that you followed these steps?

- Remove the workaround from https://github.com/docker/for-win/issues/12573#issuecomment-1047780068 (i.e. remove `features-overrides.json`)
- Install the build 75084 from https://desktop-stage.docker.com/win/main/amd64/75084/Docker%20Desktop%20Installer.exe
- Start Docker Desktop once the installation done
- No whale icon appears, even Docker is running (i.e. `docker ps` outputs something)

**Comment by lukethegeek** [comment]
@trungutt yes, those exact steps and outcomes.

Also yes, despite there being no GUI / whale icon, Docker appears to be running and I am able to interact with docker on the command line - both in powershell and a WSL2 Ubuntu shell. I just tested spinning up a mongo server & mongo express web interface via docker-compose and it's all working correctly, containers are displayed in the output of `docker ps`.

**Comment by adpe** [comment]
I've encounter the same problem @lukethegeek

**Comment by benlumia007** [comment]
@adpe can you also restart your computer and see if the docker tray icon pops up as well.

**Comment by benlumia007** [comment]
so after doing some test again, after restarting the system, the icon doesn't show up, so this means that everytime i need to do a `wsl --shutdown` to bring back the icon. is this another bug. i never had this issue.

**Comment by benlumia007** [comment]
i believe i fixed this by uninstalling ubuntu 20.04 LTS for wsl and installing it, and now it works.

**Comment by EdgarGrimbergNOV** [comment]
Here's the diagnostic from my computer. manifesting the same issue: 373C18BF-F79E-4A45-AC15-C10F9705AC77/20220228101106

**Comment by trungutt** [comment]
@lukethegeek @adpe @EdgarGrimbergNOV  Please let me try again one more time. Could you please try this build? Obviously the same procedure described in https://github.com/docker/for-win/issues/12573#issuecomment-1050332792

https://desktop-stage.docker.com/win/main/amd64/75257/Docker%20Desktop%20Installer.exe

If it still happens, please upload your diagnostics ID like described above. Thanks a lot for that!

**Comment by lukethegeek** [comment]
@trungutt No luck I'm afraid.

Same thing happens with 75257 - Docker is running in the background (`docker ps` works) but no GUI or systray icon.

Diagnostics ID:     7A765C8E-24A2-4E4A-8D66-F5C9965122C9/20220302190303 (uploaded)

More than happy to keep trying further builds as long as they keep coming, it's really no problem :)

**Comment by trungutt** [comment]
@lukethegeek Thanks a lot. Could you try this build? https://desktop-stage.docker.com/win/main/amd64/75278/Docker%20Desktop%20Installer.exe

When this build fails, could you please try `wsl --setdefault docker-desktop` and then restart Docker Desktop. Please upload your diagnostics ID after all this

**Comment by lukethegeek** [comment]
@trungutt initially it failed like before, then after running `wsl --setdefault docker-desktop` and restarting, Docker GUI started up just fine.

Diagnostics ID:     7A765C8E-24A2-4E4A-8D66-F5C9965122C9/20220302225922 (uploaded)

**Comment by lukethegeek** [comment]
@trungutt something else in case it helps:-

If I change the default wsl distro back to my Ubuntu distro, Docker GUI fails to start.

However if I then issue a `wsl --shutdown`, Docker GUI then opens along with a 'Docker Desktop WSL 2 backend has stopped unexpectedly. Do you want to restart it?' prompt. If I click 'Yes' to restart it, Docker GUI then stays open, however the moment I click on a running container, the GUI immediately disappears along with the systray icon. This process is repeatable.

I did another diagnostics after doing this process in case it triggers some kind of helpful error in the logs:-

Diagnostics ID:     7A765C8E-24A2-4E4A-8D66-F5C9965122C9/20220302235431 (uploaded)

**Comment by adpe** [comment]
Hi @trungutt 

Using the build mentioned in https://github.com/docker/for-win/issues/12573#issuecomment-1057463886 it didn't show up after a restart. Running `wsl --setdefault docker-desktop` after the update and restarting my machine worked (rebooted twice and interacted with containers between)! Therefore, I'll close the issue #12575.

Thanks for all your efforts,
Adrian

**Comment by trungutt** [comment]
@lukethegeek It's a bit hard to tell what is the exact problem when the GUI fails to start with your Ubuntu distro to be honest.

However, I tried to get into the GUI disappearance problem when clicking on a container and it looked like a real bug from our side. The fix will be deployed for the next release, but I can give you a temporary build that could fix the problem. https://desktop-stage.docker.com/win/main/amd64/75325/Docker%20Desktop%20Installer.exe Please let me know if it works

**Comment by netcore-jroger** [comment]
@trungutt 

> @lukethegeek It's a bit hard to tell what is the exact problem when the GUI fails to start with your Ubuntu distro to be honest.
> 
> However, I tried to get into the GUI disappearance problem when clicking on a container and it looked like a real bug from our side. The fix will be deployed for the next release, but I can give you a temporary build that could fix the problem. https://desktop-stage.docker.com/win/main/amd64/75325/Docker%20Desktop%20Installer.exe Please let me know if it works

always not working.

**Comment by trungutt** [comment]
@netcore-jroger Could you be more specific?

**Comment by netcore-jroger** [comment]
> @netcore-jroger Could you be more specific?

After I upgrade to v4.5.1, the DockerDesktop app always running in background.
OS: Win10 Professional 21H2
OS Internal Version: 19044.1526
```shell
$ wsl -l
适用于 Linux 的 Windows 子系统分发版:
Ubuntu-20.04 (默认)
docker-desktop
docker-desktop-data

$ uname -a
Linux ATP-YQYF-127 5.10.60.1-microsoft-standard-WSL2 #1 SMP Wed Aug 25 23:20:18 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by benlumia007** [comment]
@netcore-jroger did you try `wsl --setdefault docker-desktop` <-- should be  prority , if tht doesn't work, try delete Ubuntu or do a reset.

**Comment by gpettey** [comment]
And all of a sudden this morning the tray icon showed up. System has not been rebooted for four days, Docker is still 4.5.1.

Diag E47DA4F9-6DED-433F-B321-4B3102BDA456/20220304155430

**Comment by netcore-jroger** [comment]
> @netcore-jroger did you try `wsl --setdefault docker-desktop` <-- should be prority , if tht doesn't work, try delete Ubuntu or do a reset.

thanks. when I use `wsl --set-default docker-desktop`,then start docker-desktop app. worked fine for me.

**Comment by lukethegeek** [comment]
@trungutt sorry for the delay in getting back to you.

75325 has fixed the closing GUI bug - many thanks. At some point I'll try a fresh WSL Ubuntu install to see if that fixes Docker opening issue.

**Comment by trungutt** [comment]
Thank you all a lot for your patience

**Comment by mat007** [comment]
We have released Docker Desktop 4.6.0, with a fix for this issue, see https://docs.docker.com/desktop/windows/release-notes/

**Comment by raduzaharia** [comment]
test

**Comment by raduzaharia** [comment]
The method that work for me was Deleting in %APPDATA% the Docker and DockerDesktop folders then unistall then install again . restart pc and launching docker desktop ( and all went smooth )

**Comment by StefanScherer** [comment]
There is a follow-up ticket #12622 that we're looking into right now.
Some users here might had the same issue, who used the workaround to set the `docker-desktop` distro as default WSL2 distro. 
In the meantime to be able to set another WSL2 distro as default, you could just create an empty file in the Windows users home directory `.kube\config` until we ship a fix for it.
